File: university-study/算法/2022软院码拉松/code/收房租/Main/src/Main.java
import java.util.Scanner;

class Main {
    public static int rob(int[] nums) {
        int sum = 0 , count = nums.length;
        while (count > 0){
            int i = maxTag(nums);
            sum += nums[i];
            nums[i] = 0;
            count--;
            if(i==0 && nums[i+1] != 0){
                nums[i+1] = 0;
                count--;
            } else if (i == nums.length - 1 && nums[i-1] != 0) {
                nums[i-1] = 0;
                count--;
            }else if(i>0 && i<nums.length-1 ){
                if(nums[i-1]!=0){
                    nums[i-1] = 0;
                    count--;
                }
                if(nums[i+1]!= 0){
                    nums[i+1] = 0;
                    count--;
                }
            }
        }
        return sum;
    }

    private static int maxTag(int[] nums) {
        int tag = 0;
        for (int i = 1; i < nums.length; i++) {
            if(nums[i]>nums[i-1]){
                tag = i;
            }
        }
        return tag;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] ans = new int[n];
        for (int i = 0; i < n; i++) {
            ans[i] = scanner.nextInt();
        }
        System.out.println(rob(ans));
    }
}

File: university-study/算法/2022软院码拉松/code/收房租/C/main.c
#include<stdio.h>

int rob(int* nums,int n) {
    int sum = 0 , count = n;
    while (count > 0){
        int i = MaxTag(nums,n);
        sum += nums[i];
        nums[i] = 0;
        count--;
        if(i==0 && nums[i+1] != 0){
            nums[i+1] = 0;
            count--;
        } else if (i == n - 1 && nums[i-1] != 0) {
            nums[i-1] = 0;
            count--;
        }else if(i>0 && i<n-1 ){
            if(nums[i-1]!=0){
                nums[i-1] = 0;
                count--;
            }
            if(nums[i+1]!= 0){
                nums[i+1] = 0;
                count--;
            }
        }
    }
    return sum;
}

int MaxTag(int* nums,int n) {
    int tag = 0;
    for (int i = 1; i < n; i++) {
        if(nums[i]>nums[i-1]){
            tag = i;
        }
    }
    return tag;
}
int main(){
    int n;
    scanf("%d",&n);
    int nums[n];
    for(int i=0;i<n;i++){
        scanf("%d",&nums[i]);
    }
    printf("%d",rob(nums,n));
    return 0;
}
